service/core/client/src/main/scala/com/lightbend/lagom/internal/client/WebSocketClient.scala

```diff
@@ -399,3 +401,12 @@ object WebSocketClient {
 class WebSocketException(s: String, th: Throwable) extends java.io.IOException(s, th) {
   def this(s: String) = this(s, null)
 }
+
+@Singleton
+class WebSocketClientConfig @Inject() (val configuration: Config) {
+
+  @deprecated(message = "prefer `config` using typesafe Config instead", since = "1.4.0")
+  def this(configuration: Configuration) = this(configuration.underlying)
+
+  val config: Config = configuration.getConfig("lagom.client.websocket")
```

Comment:
Hi @datalchemist I think there are a couple of problems with this approach:

1. I don't think we should add a method and deprecate it all at once. I think the `def this(configuration: Configuration)`is not necessary
2. `WebSocketClientConfig` should not expose `val config:Config` as it's leaking all other settings to users of this class. Instead, `WebSocketClientConfig` should only have a `val maxFrameLength: Int` field.

You can have a look at other similar parts of Lagom like [Broker API's `ClientConfig`](https://github.com/lagom/lagom/blob/88102075090a91f8ad37655d599e6bdae9bedb73/service/core/kafka/client/src/main/scala/com/lightbend/lagom/internal/broker/kafka/KafkaConfig.scala#L30-L42),  `KafkaConfig`, etc...

Reply:
Hi @ignasi35 ,

Thanks for your feedback! 

Actually, I reproduced without thinking enough the pattern used for the [`CircuitBreakerConfig`](https://github.com/lagom/lagom/blob/3a544ab18d0f50d03c572c19ef6294aadf4c2693/service/core/client/src/main/scala/com/lightbend/lagom/internal/client/CircuitBreakersPanelInternal.scala#L105). I should have looked at other cases because it looked a bit weird to me. The situation is a bit different as there can be different named configs but now that I have seen your comments and other good typed config examples such as `ReadSideConfig`, the `CircuitBreakerConfig` looks not very  nicely implemented...

So, regarding your remarks:
1. Yes, indeed, that does not make much sense to provide a deprecated method for a new class...
2. Indeed. I fully agree

 Now I am wondering what is the best pattern. For the [KafkaConfig](https://github.com/lagom/lagom/blob/88102075090a91f8ad37655d599e6bdae9bedb73/service/core/kafka/client/src/main/scala/com/lightbend/lagom/internal/broker/kafka/KafkaConfig.scala#L30-L42) it is implemented using a sealed trait and a private implementation while for the [`ReadSideConfig`](https://github.com/lagom/lagom/blob/e9f9e82490218756c92cc6e9e9c3d7c116eb81ae/persistence/core/src/main/scala/com/lightbend/lagom/internal/persistence/ReadSideConfig.scala) it is implemented with a case class. 

On my side, I would have opted for the case class solution which seems to make sense as these typed Config classes are essentially records of simple values (with a constructor taking a `Config`)...Do you think there is a good reason for not using a case class and/or for hiding the implementation? This can prevent direct construction, i.e. without using a `Config` object, but then, is there a reason for that?